package/MDAnalysis/auxiliary/base.py

```diff
@@ -0,0 +1,187 @@
+import numpy as np
+
+class AuxReader(object):
+    """ Base class for auxiliary readers
+    Does iteration and aligning between trajectory ts/auxiliary step
+    Actual reading/parsing of data to be handled by individual readers.
+    """
+      
+    # TODO deal with changing/different units
+
+    def __init__(self, auxnames, represent_ts_as='closest', 
+                 cutoff=None, dt=None, time_first_col=False, initial_time=None):
+        # TODO default auxnames; allow to name each column + pick which to keep
+
+        self.names = auxnames
+        self.represent_ts_as = represent_ts_as
+        self.cutoff = cutoff ## UNITS?
+        self.time_first_col = time_first_col  ## what if time in another column?
+
+        self.ts_data = None
+        self.ts_rep = None
+
+        self.step = 0
+        self._read_next_step()
+
+        # set initial time; default to 0
+        if time_first_col:
+            init_t = self.time
+        elif initial_time:
+            init_t = intial_time
+        else:
+            init_t = 0 # TODO - warn using default?
+        self.initial_time = init_t
+
+        # set dt (assuming constant!); default to 1 ps.
+        if time_first_col:
+            self._read_next_step()
+            dt = self.time - self.initial_time
+            self.go_to_first_step()
+        elif dt:
+            self.dt = dt
+        else:
+            init_t = 1 # TODO - warn using default; (check units!)
+        self.dt = dt
+        
+
+    def next(self):
+        """ Move to next step of data """
+        return self._read_next_step()
+
+    def __next__(self):
+        """ Move to next step of data """
+        return self.next()
+
+    def __iter__(self):
+        self._restart()
+        return self
+
+    def _restart(self):
+        """ Reset back to start; calling next should read in first step """
+        # Overwrite when reading from file to seek to start of file
+        self.step = 0
```

Comment:
Step count should be 0 based, like everything else. So here `self.step` should be set to -1.